src/icp_canister_backend/src/lib.rs

```diff
@@ -208,6 +209,37 @@ pub fn get_deposit(id: u64) -> Option<types::Deposit> {
 pub fn get_pool_state() -> PoolState {
     POOL_STATE.with(|state| state.borrow().get().clone())
 }
+#[ic_cdk::query]
+pub fn get_deposits_rewards(deposit_ids: Vec<u64>) -> Vec<Nat> {
+    let current_accumulated_reward = ACCUMULATED_REWARD_PER_SHARE.with(|cell| cell.borrow().get().clone().0);
+    let current_episode = get_current_episode();
+    
+    deposit_ids
+        .iter()
+        .map(|&deposit_id| {
+            DEPOSITS.with(|deposits| {
+                match deposits.borrow().get(&deposit_id) {
```

Comment:
There is an inconsistency 
In the withdraw_rewards you are writing:
if let Some(mut deposit) = deposits_ref.get(&deposit_id)
And here you are using the match
Also, you are using map/reduce here and a simple cycle in withdraw_rewards.
**You should keep a similar style throughout the code base, especially for parts that are doing a similar thing.**

In this particular case some code blocks are **the same**. 
Use withdraw_rewards as a reference and create an internal function called `collect_desposit_rewards(deposit_ids, update_deposit)`
update_deposit is a bool to check if `rewards_collected` should be updated during the call.